src/hotspot/share/gc/z/zReferenceProcessor.hpp

```diff
@@ -41,6 +41,9 @@ class ZReferenceProcessor : public ReferenceDiscoverer {
 
   ZWorkers* const      _workers;
   ReferencePolicy*     _soft_reference_policy;
+  // Judge whether the _soft_reference_policy is AlwaysClearPolicy or not,
+  // use this extra field to avoid using RTTI (runtime type information).
+  bool                 _always_clear_policy;
```

Comment:
This name doesn't capture that it has something to do with soft references. I think _clear_all_soft_refs is a better name.

Reply:
Renamed.